azurerm/resource_arm_bot_channel_email_test.go

```diff
@@ -0,0 +1,170 @@
+	config2 := testAccAzureRMBotChannelEmail_basicUpdate(ri, testLocation())
+	resourceName := "azurerm_bot_channel_email.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMBotChannelEmailDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMBotChannelEmailExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"email_password",
+				},
+			},
+			{
+				Config: config2,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMBotChannelEmailExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"email_password",
+				},
+			},
+		},
+	})
+}
+
+func testCheckAzureRMBotChannelEmailExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		botName := rs.Primary.Attributes["bot_name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Bot Channel Email")
+		}
+
+		client := testAccProvider.Meta().(*ArmClient).bot.ChannelClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := client.Get(ctx, resourceGroup, botName, string(botservice.ChannelNameEmailChannel))
+		if err != nil {
+			return fmt.Errorf("Bad: Get on botChannelClient: %+v", err)
+		}
+
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Bad: Bot Channel Email %q (resource group: %q / bot: %q) does not exist", name, resourceGroup, botName)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMBotChannelEmailDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).bot.ChannelClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_bot_channel_email" {
+			continue
+		}
+
+		botName := rs.Primary.Attributes["bot_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := client.Get(ctx, resourceGroup, botName, string(botservice.ChannelNameEmailChannel))
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Bot Channel Email still exists:\n%#v", resp.Properties)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMBotChannelEmail_basicConfig(rInt int, location string) string {
+	template := testAccAzureRMBotChannelsRegistration_basicConfig(rInt, location)
```

Comment:
this test fails because:

>         Error: Error issuing create request for Bot Channels Registration "acctestdf190926083626598033" (Resource Group "acctestRG-190926083626598033"): botservice.BotsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidBotData" Message="Bot is not valid. Errors: The Microsoft App ID is already registered to another bot application..  See https://aka.ms/bot-requirements for detailed requirements."

maybe we need to generate a new UUID for each bot application?